src/controllers/costcategory_controller.go

```diff
@@ -0,0 +1,119 @@
+package controllers
+
+import (
+	"github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/expense_errors"
+	"github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/managers"
+	"github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/models"
+	"github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/repositories"
+	"github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/utils"
+	"github.com/google/uuid"
+)
+
+// CostCategoryCtl Exposed interface to the handler-package
+type CostCategoryCtl interface {
+	CreateCostCategory(tripId *uuid.UUID, costCategoryRequest models.CostCategoryPostRequest) (*models.CostCategoryResponse, *models.ExpenseServiceError)
+	PatchCostCategory(costCategoryId *uuid.UUID, costCategoryRequest models.CostCategoryPatchRequest) (*models.CostCategoryResponse, *models.ExpenseServiceError)
+	GetCostCategoryDetails(costCategoryId *uuid.UUID) (*models.CostCategoryResponse, *models.ExpenseServiceError)
+	DeleteCostCategory(costCategoryId *uuid.UUID) *models.ExpenseServiceError
+	GetCostCategoryEntries(tripId *uuid.UUID) ([]*models.CostCategoryResponse, *models.ExpenseServiceError)
+}
+
+// CostCategoryController Cost Category Controller structure
+type CostCategoryController struct {
+	DatabaseMgr      managers.DatabaseMgr
+	CostCategoryRepo repositories.CostCategoryRepo
+}
+
+func (ccc *CostCategoryController) CreateCostCategory(tripId *uuid.UUID, createCostCategoryRequest models.CostCategoryPostRequest) (*models.CostCategoryResponse, *models.ExpenseServiceError) {
+	if utils.ContainsEmptyString(createCostCategoryRequest.Name) {
+		return nil, expense_errors.EXPENSE_BAD_REQUEST
+	}
```

Comment:
Das wolltest du in den handler auslagern, wenn ich mich richtig erinnere

Reply:
Done for next commit